trie/snap.go

```diff
@@ -0,0 +1,53 @@
+package trie
+
+import (
+	"github.com/dominant-strategies/go-quai/common"
+)
+
+// Used to send a trie node request to a peer
+type TrieNodeRequest struct{}
+
+// Used to get a trie node response from a peer
+type TrieNodeResponse struct {
+	NodeData []byte
+	NodeHash common.Hash
+}
+
+// GetTrieNode returns the trie node from a TrieNodeResponse
+func (t *TrieNodeResponse) GetTrieNode() *TrieNode {
+	node, err := decodeNode(t.NodeHash[:], t.NodeData)
+	if err != nil {
+		panic(err)
+	}
+	return &TrieNode{n: node}
+}
+
+// TrieNode is a public wrapper around a trie node that exposes
+// methods for handling trie nodes.
+type TrieNode struct {
+	n node
+}
+
+// ChildHashes returns the hashes of the children of a fullNode trie node
+func (t *TrieNode) ChildHashes() []common.Hash {
+	switch n := t.n.(type) {
+	case *fullNode:
```

Comment:
What about the short node type (or whatever it is called)? Doesn't it also have a child hash?